### PR TITLE
Prepend libs to path

### LINKED ISSUE #701: RPC proxy not working with implicit start from Rhino/Grasshopper
**Describe the bug**

Using `compas_rpc.Proxy`  as a contextmanager raises an exception. This only happens if a RPC server is not already running. (Same/similar to #265)

**To Reproduce**

**Context:** Grasshopper GHPython node or Rhino python editor.

## Sample script
```python
from compas.rpc import Proxy
from compas.geometry import Point

pt1 = Point(1, 2, 3)

with Proxy("compas.geometry", max_conn_attempts=5) as proxy:
    pt2 = proxy.Point(1, 2, 3)

if list(pt1) == pt2:
    print("Success")
```
4. Traceback
```
Message: Traceback (most recent call last):
  File "C:\Users\a\scoop\apps\miniconda3\4.7.12.1\envs\compas_rr_ts-dev\lib\site-packages\compas\rpc\dispatcher.py", line 100, in _dispatch
    module = importlib.import_module(modulename)
  File "C:\Users\a\scoop\apps\miniconda3\4.7.12.1\envs\compas_rr_ts-dev\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\a\scoop\apps\miniconda3\4.7.12.1\envs\compas_rr_ts-dev\lib\site-packages\compas\geometry\__init__.py", line 497, in <module>
    from .bbox import *  # noqa: F401 F403
  File "C:\Users\a\scoop\apps\miniconda3\4.7.12.1\envs\compas_rr_ts-dev\lib\site-packages\compas\geometry\bbox\__init__.py", line 10, in <module>
    from .bbox_numpy import *  # noqa: F401 F403
  File "C:\Users\a\scoop\apps\miniconda3\4.7.12.1\envs\compas_rr_ts-dev\lib\site-packages\compas\geometry\bbox\bbox_numpy.py", line 24, in <module>
    from compas.numerical import pca_numpy
  File "C:\Users\a\scoop\apps\miniconda3\4.7.12.1\envs\compas_rr_ts-dev\lib\site-packages\compas\numerical\__init__.py", line 109, in <module>
    from .isolines import *  # noqa: F401 F403
  File "C:\Users\a\scoop\apps\miniconda3\4.7.12.1\envs\compas_rr_ts-dev\lib\site-packages\compas\numerical\isolines\__init__.py", line 8, in <module>
    from .isolines_numpy import *  # noqa: F401 F403
  File "C:\Users\a\scoop\apps\miniconda3\4.7.12.1\envs\compas_rr_ts-dev\lib\site-packages\compas\numerical\isolines\isolines_numpy.py", line 11, in <module>
    import matplotlib.pyplot as plt
  File "C:\Users\a\scoop\apps\miniconda3\4.7.12.1\envs\compas_rr_ts-dev\lib\site-packages\matplotlib\__init__.py", line 174, in <module>
    _check_versions()
  File "C:\Users\a\scoop\apps\miniconda3\4.7.12.1\envs\compas_rr_ts-dev\lib\site-packages\matplotlib\__init__.py", line 159, in _check_versions
    from . import ft2font
ImportError: DLL load failed while importing ft2font: %1 is not a valid Win32 application.


Traceback:
  line 345, in _proxy, "C:\Users\a\AppData\Roaming\McNeel\Rhinoceros\6.0\scripts\compas\rpc\proxy.py"
  line 4, in <module>, "C:\Users\a\AppData\Local\Temp\TempScript.py"
```

**Expected behavior**
Print "Success".

**Desktop (please complete the following information):**
 - OS: Win10
 - Python version 2.7.8 (IronPython 2.7.8 (2.7.8.0) on .NET 4.0.30319.42000 (64-bit))
 - Python package manager: Anaconda
 - compas version: 0.19.1

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/_os.py [hunk 1]
```diff
@@ -221,13 +221,15 @@ def prepare_environment(env=None):
...
-        lib_bin = os.path.join(PYTHON_DIRECTORY, 'lib')
-        if os.path.exists(lib_bin) and lib_bin not in env['PATH']:
-            env['PATH'] += os.pathsep + lib_bin
+        if is_windows():
+            lib_bin = os.path.join(PYTHON_DIRECTORY, 'Library', 'bin')
+            if os.path.exists(lib_bin) and lib_bin not in env['PATH']:
+                env['PATH'] = lib_bin + os.pathsep + env['PATH']
+
+        else:
+            lib_bin = os.path.join(PYTHON_DIRECTORY, 'bin')
```
Comment (gonzalocasas) [inline src/compas/_os.py:227]: So, `os.path.join(PYTHON_DIRECTORY, 'lib')` is not needed?

File: src/compas/_os.py [hunk 2]
```diff
@@ -221,13 +221,15 @@ def prepare_environment(env=None):
...
+            if os.path.exists(lib_bin) and lib_bin not in env['PATH']:
+                env['PATH'] = lib_bin + os.pathsep + env['PATH']
```
Comment (gonzalocasas) [inline src/compas/_os.py]: Wouldn't this be preferable?

```suggestion
        if is_windows():
            lib_bin = os.path.join(PYTHON_DIRECTORY, 'Library', 'bin')
        else:
            lib_bin = os.path.join(PYTHON_DIRECTORY, 'bin')

        if os.path.exists(lib_bin) and lib_bin not in env['PATH']:
            env['PATH'] = lib_bin + os.pathsep + env['PATH']
```

File: src/compas/_os.py (hunk 1 above)
Comment (tomvanmele) [reply src/compas/_os.py:227]: on windows it doesn't exist and the equivalent of the `Library/bin` folder on mac is `bin`. on the command line none of the lib folders are added...

File: src/compas/_os.py (hunk 2 above)
Comment (tomvanmele) [reply src/compas/_os.py]: yup

**Review by gonzalocasas — APPROVED** [review]
LGTM!

**Pull request merged**